### ISSUE TITLE
Is it possible to take a screenshot/screencapture using opencv on a mac? 

### ISSUE THREAD

**Issue by spectacularcrackers** [bytedeco/javacv#1756]
I would like to take a screenshot of my mac every second using opencv. Is this possible? If so, is there an example that I could reference? 

I see the wiki document for windows (https://github.com/bytedeco/javacv/wiki/Screen-Capture-on-Windows), but I am not sure how to implement it for Mac.

**Comment by saudet** [comment]
According to the docs at https://trac.ffmpeg.org/wiki/Capture/Desktop something like this should work:
```java
  FrameGrabber grabber = new FFmpegFrameGrabber("1");
  grabber.setFormat("avfoundation");
  grabber.setFrameRate(30);
  grabber.start();`
```

**Comment by saudet** [comment]
Could you check what you get after setting the "org.bytedeco.javacpp.logger.debug" system property to "true"?

**Comment by spectacularcrackers** [comment]
I do not see the path `org.bytedeco.javacpp.logger.debug'. However, I do see the debug variable in `org.bytedeco.javacpp.tools.Logger.java`. Is this what you are referring to? If so, how do I edit the file? I am using the gradle dependency, and the external libraries are all read-only

**Comment by saudet** [comment]
Typically, calling something like `System.setProperty("org.bytedeco.javacpp.logger.debug", "true")` works.

**Comment by saudet** [comment]
I see, `FFmpegFrameGrabber.tryLoad()` isn't getting called. I mistakenly commented that out with the upgrade to FFmpeg 5.0, but it's still needed. I've fixed this in commit https://github.com/bytedeco/javacv/commit/fef1d8ee8778dca195bfca35bdf81e9031d2ed12, but we can easily work around that just by calling it explicitly manually.

**Comment by spectacularcrackers** [comment]
It works now! Thank you @saudet!

It the default pixel format (bgr24) gave an error. `[avfoundation @ 0x119fae530] Selected pixel format (bgr24) is not supported by the input device.` (for reference, I'm on a M1 Pro MacBook Pro)

I used one of the suggested supported pixel formats `grabber.setPixelFormat(avutil.AV_PIX_FMT_0RGB);`, and it worked perfectly.

**Comment by spectacularcrackers** [comment]
@saudet do you know of a more efficient way of saving the captured frames? I am currently capturing the screen frames, displaying them on a CanvasFrame and recording from the frame. Do you have an example of how to have the program capture the screen without using a CanvasFrame?

**Comment by saudet** [comment]
> @saudet do you know of a more efficient way of saving the captured frames? I am currently capturing the screen frames, displaying them on a CanvasFrame and recording from the frame. Do you have an example of how to have the program capture the screen without using a CanvasFrame?

You can stop using CanvasFrame entirely, that's just to display frames. We don't need it to record them.

> I am trying to use a while loop to continuously capture the frames. However, only one frame gets captured and the recording stops. Any ideas on what I am doing wrong?
> 
> ```
>         Frame frame;
>         while((frame = grabber.grab()) != null){
>             recorder.record(frame);
>         }
>         recorder.stop();
>         grabber.stop();
> ```
> 
> I am getting the following warnings but could not figure out how to resolve them: `[AVFoundation indev @ 0x12de64c00] Configuration of video device failed, falling back to default.` `[avfoundation @ 0x12de64980] Stream #0: not enough frames to estimate rate; consider increasing probesize` `[aac @ 0x103506660] 1 frames left in the queue on closing`

I don't know, but according to a quick search, it just sounds like a permission problem with your system, for example, see
https://github.com/kapoorlakshya/screen-recorder/issues/88#issuecomment-629139032.

**Comment by spectacularcrackers** [comment]
I also saw that thread and made sure the permissions were correctly set. The screen gets recorded for one second and the one-two frame clip gets saved. However it then terminates.

**Comment by saudet** [comment]
@spectacularcrackers There's another issue in the case of avfoundation that was fixed in commit https://github.com/bytedeco/javacv/commit/8d75c0a33c9a105c7da08e754179ac785bf6f788, see issue #1784.
Please give it a try with the snapshots: http://bytedeco.org/builds/

**Comment by spectacularcrackers** [comment]
The new snapshot fixed it. Thanks @saudet !

**Comment by saudet** [comment]
The fix has been released with JavaCV 1.5.8. Thanks for reporting!
